### PR TITLE
fix datetime displaying issue for timezone <0

### PR DESCRIPTION
**Pull request by mattwang44**
## Types of changes

* **Bugfix**

## Description

1. Failed to differentiate day 1 or day 2 for timezone within `[-12, 0)`. One may see the datetime is failed to be displayed as the image shown below:

2. On schedule page, the user with timezone within `[-12, 0)` see the dates `10/2` & `10/3`, but it's actually `10/1` & `10/2`.

image

### CODE REVIEW

File: pages/conference/_eventType/_id.vue [hunk 1]
```diff
@@ -235,22 +233,13 @@ export default {
...
-            const datetimeUtc = new Date(datetimeString)
-            const offset = new Date().getTimezoneOffset()
-            const datetime = new Date(datetimeUtc - offset)
-            return datetime
-        },
         getDateTag(beginTime) {
-            const month = beginTime.getMonth() + 1
-            const date = beginTime.getDate()
-            if (month === 10 && date === 2) {
+            const dayOneMidnight = new Date('2021-10-02 16:00:00')
```
Comment (flynnhou) [inline pages/conference/_eventType/_id.vue:237]: I'm curious why `dayOneMidnight` is this time?
Could we use ISO 8601 instead? To me, that will be more precise.

File: plugins/strings.js [hunk 2]
```diff
@@ -22,7 +22,9 @@ export const datetimeToString = (source, options = defaultDatetimeOptions) => {
         options.inputFormat || defaultDatetimeOptions.inputFormat
     const outputFormat =
         options.outputFormat || defaultDatetimeOptions.outputFormat
-    return dayjs(source, inputFormat).format(outputFormat)
+
+    const offset = new Date().getTimezoneOffset()
+    return dayjs(source, inputFormat).subtract(offset, 'm').format(outputFormat)
```
Comment (flynnhou) [inline plugins/strings.js:27]: Do we need this change in this PR? I don't see the changes using this plugin method.

Comment (mattwang44) [reply plugins/strings.js:27]: one should see the change if you change the default timezone of your system.

File: pages/conference/_eventType/_id.vue (hunk 1 above)
Comment (mattwang44) [reply pages/conference/_eventType/_id.vue:237]: This is the UTC time of the midnight in Taiwan (UTC+8).

**Review by flynnhou — APPROVED** [review]
lgtm

**Pull request merged**